compiler/src/misc.rs

```diff
@@ -16,6 +16,13 @@ pub struct MastermindConfig {
 	pub optimise_constants: bool,
 	// TODO: recursively prune if statements/loops if they do nothing
 	pub optimise_empty_blocks: bool,
+	// Memory Allocation Method
+	//'1D Mastermind'  0
+	// '2D Mastermind - Spiral'  1
+	// '2D Mastermind - Tiles'  2
+	// '2D Mastermind - Nearest' 3
+	pub memory_allocation_method: u8,
```

Comment:
I wonder if this could be an enum or string or something, I guess it might get fucky when serialising between js and wasm.